src/telemetry.ts

```diff
@@ -36,4 +49,80 @@ export class NullTelemetryReporter implements ITelemetryReporter {
 
 }
 
+export const DefaultTelemetryInterval = 10000;
+
+export class BatchTelemetryReporter implements ITelemetryReporter {
+    private _eventBuckets: {[eventName: string]: any};
+    private _timer: NodeJS.Timer;
+
+    public constructor(private _telemetryReporter: TelemetryReporter, private _cadence: number = DefaultTelemetryInterval) {
+        this.reset();
+        this.setup();
+    }
+
+    public reportEvent(name: string, data?: any): void {
+        if (!this._eventBuckets[name]) {
+            this._eventBuckets[name] = [];
+        }
+
+        this._eventBuckets[name].push(data);
+    }
+
+    public setupEventHandler(_sendEvent: (event: DebugProtocol.Event) => void): void {
+        // no-op
```

Comment:
It'll be surprising for customers of this class if this is a no-op. It might be clearer if this class doesn't implement ITelemetryReporter and doesn't have this method.

Reply:
Very valid point.